packages/core/test/helpers/namespace.spec.ts

```diff
@@ -0,0 +1,185 @@
+import { expect } from 'chai';
+import {
+    type NamespaceBuilderParams,
+    createNamespaceStrategy,
+    defaultNamespaceBuilder,
+    defaultNoMatchHandler,
+} from '@stylable/core/dist/helpers/namespace';
+
+describe('createNamespaceStrategy', () => {
+    it('should return smallest namespace for the same file', () => {
+        const resolveNamespace = createNamespaceStrategy({
+            hashFn: () => '1',
+            getPackageInfo: () => ({ name: 'package', version: '0.0.0', dirPath: '/package' }),
+            normalizePath: (dirPath, filePath) => filePath.replace(dirPath, ''),
+        });
+
+        expect(resolveNamespace('x', '/package/x.st.css')).to.equal('x');
+        expect(resolveNamespace('x', '/package/x.st.css')).to.equal('x');
+    });
+    it('should return smallest namespace for the same file with prefix', () => {
+        const resolveNamespace = createNamespaceStrategy({
+            prefix: 'test-',
+            hashFn: () => '1',
+            getPackageInfo: () => ({ name: 'package', version: '0.0.0', dirPath: '/package' }),
+            normalizePath: (dirPath, filePath) => filePath.replace(dirPath, ''),
+        });
+
+        expect(resolveNamespace('x', '/package/x.st.css')).to.equal('test-x');
+        expect(resolveNamespace('x', '/package/x.st.css')).to.equal('test-x');
+    });
+    it('should salt the hash', () => {
+        let usedInput: string | undefined;
+        const resolveNamespace = createNamespaceStrategy({
+            hashSalt: '__SALT__',
+            hashFn: (input) => {
+                usedInput = input;
+                return '1';
+            },
+            getPackageInfo: () => ({ name: 'package', version: '0.0.0', dirPath: '/package' }),
+            normalizePath: (dirPath, filePath) => filePath.replace(dirPath, ''),
+        });
+
+        expect(resolveNamespace('x', '/package/x.st.css')).to.equal('x');
+        expect(usedInput).to.equal('__SALT__package@0.0.0//x.st.css');
+    });
+    it('should call buildNamespace hook', () => {
+        let usedOptions: NamespaceBuilderParams | undefined;
+        const resolveNamespace = createNamespaceStrategy({
+            hashFn: () => '1',
+            getPackageInfo: () => ({ name: 'package', version: '0.0.0', dirPath: '/package' }),
+            normalizePath: (dirPath, filePath) => filePath.replace(dirPath, ''),
+            buildNamespace(options) {
+                usedOptions = options;
+                return defaultNamespaceBuilder(options);
```

Comment:
return hardcoded result {namespace: "customNS", hashPart: "12345"} to show the function controls the final namespace

Reply:
not sure why

Reply:
to emphasis the expectation, but it's not that important.